### ISSUE TITLE
Read-only prevents boa on structs.

### ISSUE THREAD

**Issue by bjourne** [factor/factor#1722]
This should work:

``` factor
STRUCT: hi { a uint read-only } ;
10 hi <struct-boa>
Attempting to define >>a outside of a compilation unit

Type :help for debugging help.
```

**Comment by mrjbq7** [comment]
Hah, nice bug find!  It seems ``<struct-boa>`` is implemented by first allocating memory for the struct and then using the setter words to fill in the values...

**Comment by mrjbq7** [comment]
Fixed in 147e9ec54ef7c6c17d0e628e2d7cc8681e2f750f.
